common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -853,3 +853,79 @@ def setUp(self):
+@ddt.ddt
+class WordCloudTest(UniqueCourseTest):
+    """
+    Tests the Word Cloud.
+    """
+    USERNAME = "STUDENT_TESTER"
+    EMAIL = "student101@example.com"
+
+    def setUp(self):
+        super(WordCloudTest, self).setUp()
+
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+
+        self.studio_course_outline = StudioCourseOutlinePage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+
+        # Install a course
+        course_fix = CourseFixture(
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        )
+        # Set word cloud value against advanced modules in advanced settings
+        course_fix.add_advanced_settings({
+            "advanced_modules": {"value": ["word_cloud"]},
+        })
+
+        course_fix.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section 1').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection 1').add_children(
+                    XBlockFixtureDesc('vertical', 'Test Unit').add_children(
+                        XBlockFixtureDesc(
+                            'word_cloud', 'advanced WORDCLOUD'
+                        )
+                    )
+                )
+            )
+        ).install()
+
+    def test_word_cloud_is_rendered_with_empty_result(self):
+        """
+        Scenario: Word Cloud component in LMS is rendered with empty result
+        Given the course has a Word Cloud component
+            Then I view the word cloud and it has rendered
+        When I press the Save button
+            Then I see the empty result
+        """
+        auto_auth(self.browser, self.USERNAME, self.EMAIL, False, self.course_id)
+        self.courseware_page.visit()
+        self.assertTrue(self.courseware_page.is_word_cloud_rendered)
+        self.courseware_page.save_word_cloud()
+        self.assertEqual(self.courseware_page.word_cloud_answer_list, '')
+
+    def test_word_cloud_is_rendered_with_result(self):
+        """
+        Scenario: Word Cloud component in LMS is rendered with result
+        Given the course has a Word Cloud component
+            Then I view the word cloud and it has rendered
+        When I fill inputs
+        And I press the Save button
+            Then I see the result with words count
+        """
+        expected_data = ['test_wordcloud1', 'test_wordcloud2', 'test_wordcloud3', 'test_wordcloud4', 'test_wordcloud5']
+        auto_auth(self.browser, self.USERNAME, self.EMAIL, False, self.course_id)
+        self.courseware_page.visit()
+        self.assertTrue(self.courseware_page.is_word_cloud_rendered)
+        self.courseware_page.input_word_cloud('test_wordcloud')
+        self.courseware_page.save_word_cloud()
+        self.assertEqual(len(self.courseware_page.word_cloud_answer_list), 5)
```

Comment:
This assert is unnecessary because the below assert statement implicitly checks the length.